src/auth/auth.service.ts

```diff
@@ -0,0 +1,36 @@
+import { BadRequestException, Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { UsersService } from '../users/users.service';
+import * as bcrypt from 'bcrypt';
+import { AuthenticateDTO, GoogleDTO, LoginDTO, RegisterDTO } from './auth.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService, private jwtService: JwtService) {}
+
+  async validateUser(userData: AuthenticateDTO): Promise<any> {
+    const { email, password } = userData;
+    const user = await this.usersService.findByEmail(email);
+
+    if (user) {
+      if (await bcrypt.compare(password, user.password)) {
+        const { password, ...result } = user;
+        return result;
+      } else return new HttpException('Invalid credentials', HttpStatus.UNAUTHORIZED);
+    } else return new HttpException('User does not exist', HttpStatus.NOT_FOUND);
+  }
+
+  async googleLogin(user: GoogleDTO) {
+    if (!user) throw new BadRequestException();
```

Comment:
`BadRequestException` w serwisie to też nie jest dobry pomysł, za to jest odpowiedzialny kontroler.